OpenSim/Common/Reporter.h

```diff
@@ -163,10 +163,24 @@ OpenSim_DECLARE_CONCRETE_OBJECT_T(TableReporter_, InputT, Reporter<InputT>);
     TableReporter_() = default;
     virtual ~TableReporter_() = default;
 
+    /** Retrieve the report.                                                  */
```

Comment:
@klshrinidhi and I discussed the clear-on-read philosophy. The potential issue is that some other process might want/need to read the report without clearing, in which case we'd need a sniff/read-without-clearing method. Throwing an exception if `time_to_append < time_last_written` and telling the user to call `clearReport()` if they're sure they want to overwrite the data seems like a good solution.